numpy/core/tests/test_multiarray.py

```diff
@@ -2497,6 +2497,35 @@ def test_diagonal_memleak(self):
         if HAS_REFCOUNT:
             assert_(sys.getrefcount(a) < 50)
 
+    def test_size_zero_memleak(self):
+        # Regression test for issue 9615
+
+        orig = np.zeros((6*4+1,), dtype='int8', order='c')[1:]
+        orig.dtype = 'f4'
+        orig[:] = 5
+        copy = np.nditer(orig, [], [['readwrite', 'updateifcopy', 'aligned']]).operands[0]
```

Comment:
Hmm, we have been trying to get rid of updateifcopy. @mattip ?

Reply:
this will be eliminated shortly